tcms/testcases/tests/test_views.py

```diff
@@ -228,6 +232,95 @@ def _assertTestCase(self, test_case):
+    def setUpTestData(cls):
+        cls.tester = UserFactory()
+        cls.case_status_confirmed = TestCaseStatus.objects.get(name='CONFIRMED')
+        cls.case_status_proposed = TestCaseStatus.objects.get(name='PROPOSED')
+
+        cls.product = ProductFactory()
+        cls.version = VersionFactory(value='0.1', product=cls.product)
+
+        cls.plan = TestPlanFactory(
+            author=cls.tester,
+            product=cls.product,
+            product_version=cls.version)
+
+        cls.case = TestCaseFactory(
+            author=cls.tester,
+            default_tester=None,
+            reviewer=cls.tester,
+            case_status=cls.case_status_confirmed,
+            plan=[cls.plan])
+        cls.case.save()  # will generate history object
+        cls.summary = 'summary'
+
+        cls.post_data = {
+            'author': cls.tester.pk,
+            'summary': cls.summary,
+            'default_tester': cls.tester.pk,
+            'product': cls.case.category.product.pk,
+            'category': cls.case.category.pk,
+            'case_status': cls.case_status_confirmed.pk,
+            'priority': cls.case.priority.pk,
+            'text': 'some text description',
+            'script': 'some script',
+            'arguments': 'args1, args2, args3',
+            'requirement': 'requirement',
+            'extra_link': 'http://somelink.net',
+            'notes': 'notes',
+
+            'email_settings-0-auto_to_case_author': 'on',
+            'email_settings-0-auto_to_run_manager': 'on',
+            'email_settings-0-auto_to_case_run_assignee': 'on',
+            'email_settings-0-auto_to_case_tester': 'on',
+            'email_settings-0-auto_to_run_tester': 'on',
+            'email_settings-0-notify_on_case_update': 'on',
+            'email_settings-0-notify_on_case_delete': 'on',
+            'email_settings-0-cc_list': 'info@example.com',
+            'email_settings-0-case': '',
+            'email_settings-0-id': cls.case.emailing.pk,
+            'email_settings-TOTAL_FORMS': '1',
+            'email_settings-INITIAL_FORMS': '1',
+            'email_settings-MIN_NUM_FORMS': '0',
+            'email_settings-MAX_NUM_FORMS': '1',
+        }
+
+        super().setUpTestData()
+
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+
+    def verify_get_without_permission(self):
+        response = self.client.get(self.url)
+        redirect_url = "{0}?next={1}".format(
+            reverse('tcms-login'), reverse('testcases-new')
+        )
+        self.assertRedirects(response, redirect_url)
+
+    def verify_post_with_permission(self):
+        response = self.client.post(self.url, self.post_data)
+        test_case = TestCase.objects.get(summary=self.summary)
+        redirect_url = reverse('testcases-get', args=[test_case.pk])
+        self.assertRedirects(response, redirect_url, status_code=302, target_status_code=302)
```

Comment:
you can post with `follow=True` which will follow all redirects and then you can assert that the response is the page displaying a test case with the expected values. 

Checking some well known elements/fields on the page is a good idea.